IoT/Files/Parameters/IoTHubReuseParameters.json

```diff
@@ -0,0 +1,193 @@
+        "attributes": "primary;"
+      },
+      "dataSourceId": "listExistingHubs",
+      "visibleRule": "reuseHub == true"
+    },
+    {
+        "id": "iotHubName",
+        "type": "string",
+        "description": "Name of IoT Hub to be created",
+        "Name": "IoT Hub name",
+        "inputMode": "TextBox",
+        "groupId": "iotHub",
+        "isRequired": true,
+        "defaultValue": "{{{system.projectName}}}-iothub",
+        "properties": {
+            "attributes": "primary;"
+        },
+        "staticValidation": {
+            "pattern": "[a-zA-Z0-9\\-]*",
+            "errorMessage": "IotHub name can contain only alphanumeric and hyphen and must start/end with alphanumeric. Should be 3 to 50 chars long",
+            "minLength": 3,
+            "maxLength": 50
+        },
+        "dynamicValidations": [
+            {
+                "dataSourceId": "checkIotHubAvailability",
+                "errorMessage": "IoT Hub with this name already exists"
+            }
+        ],
+        "visibleRule": "reuseHub == false"
+    },
+    {
+      "id": "iotHubSku",
+      "type": "string",
+      "description": "Details of cost and compute resource associated with the IoT Hub",
+      "name": "Pricing tier",
+      "inputMode": "Combo",
+      "groupId": "iotHub",
+      "isRequired": true,
+      "possibleValues": [
+        {
+          "value": "S1",
+          "displayValue": "S1 Standard"
+        },
+        {
+          "value": "S2",
+          "displayValue": "S2 Standard"
+        },
+        {
+          "value": "S3",
+          "displayValue": "S3 Standard"
+        },
+        {
+          "value": "B1",
+          "displayValue": "B1 Basic"
+        },
+        {
+          "value": "B2",
+          "displayValue": "B2 Basic"
+        },
+        {
+          "value": "B3",
+          "displayValue": "B3 Basic"
+        },
+        {
+          "value": "F1",
+          "displayValue": "F1 Free"
+        }
+      ],
+      "defaultValue": "S1",
+        "properties": {
+            "summaryText": "{{#equals}}{{inputs.reuseHub}} 'false' true '{{{inputs.iotHubSku}}}'{{/equals}}"
+        },
+      "visibleRule": "reuseHub == false"
+    },
+    {
+      "id": "location",
+      "type": "string",
+      "description": "Location of the IoT Hub.",
+      "name": "Location",
+      "inputMode": "Combo",
+      "groupId": "iotHub",
+      "isRequired": true,
+      "dataSourceId": "iotHubLocations",
+      "defaultValue": "South Central US",
+      "properties": {
+        "attributes": "primary;"
+      },
+      "visibleRule": "reuseHub == false"
+    },
+    {
+      "id": "existingHubResourceGroupLocation",
+      "type": "string",
+      "description": "Location of the resource group contianing existing IoT Hub.",
+      "Name": "Resource group Location",
+      "inputMode": "TextBox",
+      "groupId": "iotHub",
+      "isRequired": true,
+      "dataSourceId": "getResourceGroupLocation",
```

Comment:
this value is read-only? is it necessary

Reply:
this is necessary as we need to know resource group of existing hub. resource group name is used to assign appropriate permission to SPN for deploying to this hub.